src/interfaces/events/handlers/events-get.ts

```diff
@@ -0,0 +1,51 @@
+import type { DidResolver } from '../../../index.js';
+import type { EventsGetMessage } from '../types.js';
+import type { GetEventsOptions } from '../../../event-log/event-log.js';
+import type { MethodHandler } from '../../types.js';
+import type { Event, EventLog } from '../../../event-log/event-log.js';
+
+
+import { authenticate } from '../../../core/auth.js';
+import { EventsGet } from '../messages/events-get.js';
+import { MessageReply } from '../../../core/message-reply.js';
+
+type HandleArgs = {tenant: string, message: EventsGetMessage};
+
+export class EventsGetHandler implements MethodHandler<Event> {
+  constructor(private didResolver: DidResolver, private eventLog: EventLog) {}
+
+  public async handle({ tenant, message }: HandleArgs): Promise<MessageReply<Event>> {
+    let eventsGet: EventsGet;
+
+    try {
+      eventsGet = await EventsGet.parse(message);
+    } catch (e) {
+      return MessageReply.fromError(e, 400);
+    }
+
+    try {
+      await authenticate(message.authorization, this.didResolver);
+
+      if (eventsGet.author !== tenant) {
+        //! FIXME: think of better error message
```

Comment:
haha fair. haven't been able to think of a good error message yet! all ears

Reply:
Honestly I thought whatever you got is good enough, author must be tenant, what more do you want! = ]

Reply:
most of the other handlers do something like `eventsGet.authorize(tenant)`

should we add a `EventsGet.prototype.authorize` to do this instead?
```
public async authorize(tenant: string): Promise<void> {
  // if author/requester is the same as the target tenant, we can directly grant access
  if (this.author === tenant) {
    return;
  } else {
    throw new Error('message failed authorization');
  }
}
```

Reply:
switched to using `authorize` instead of repeating the code. it's a wee bit strange imo that the message classes have `authorize` methods because `tenant` is a bit out of context there